src/models/ActivitySchema.ts

```diff
@@ -1,11 +1,18 @@
 import * as Joi from "joi";
+import {ActivityType} from "./Activity";
+import {waitReasons} from "./Activity";
 
 export const ActivitySchema = Joi.object().keys({
-    activityType: Joi.any().only([ "visit", "wait" ]).required(),
+    parentId: Joi.string().optional(),
+    activityType: Joi.any().only([ "visit", "wait", ActivityType.unaccountableTime ]).required(),
```

Comment:
can you please move the values in a constant, similar as you did with the waitReasons?

Reply:
done.